[10.0] Port l10n_ch_hr_payroll

open-net-sarl opened the pull request:
- Added the configuration wizard
- Added new rules and changed some of the old ones
- Added 'Source Tax',  OBP contract and occupation rate to the employee contract
- Added working rate to the payslip
- Added translations in French, Italian and German
- Added tests for the new features

open-net-sarl committed:
Improvements on CH salary rules.
New features:
- Wizard to configurate accounting and percentages on rules.
- Translations of rule names and notes
- Working percentage and occupation rate added to compute basic amount of wage.
- Configuration saved into company
- LPP contracts
- Added two fields to retrieve percentage and amount for python computed rules

open-net-sarl committed:
Added test files
Added Rules BASIC_CH, NET_CH, GROSS_CH

open-net-sarl committed:
Added More tests

open-net-sarl committed:
- Added more testes
- Added Rule 'Source Tax' to the rules and to the contract

open-net-sarl commented:
Work in progress

open-net-sarl committed:
-Resolved bad query for wage fulltime and occupation rate

open-net-sarl committed:
- Added security access rules for the model lpp.contract

open-net-sarl committed:
- Fixed security access rules for lpp.contract

open-net-sarl committed:
- Fixed permitions on configuration wizard and lpp contract

open-net-sarl committed:
- Fixed wrong account assigned to NET CH
- Added filters for each account in wizard view

open-net-sarl committed:
- Fixed translation for "OBP Contract" in the wizard
- Fixed wizard test
- Fixed contract view

open-net-sarl commented:
Work done

didierdonze commented:
Thank you for your work!
- It seems you added two columns (python rate and python amount). The 2 columns should be calculated as well on the view "Payslip Computation Details" and on the standard PDF payslip

open-net-sarl commented:
@didierdonze The two columns (python rate and python amount) are present on the "Payslip" view, they replace the originals columns "Rate" and "Amount".
The standard PDF payslip its not us that are working on, is C2C.

didierdonze commented:
- I meant on the view "Payslip Computation Details". See the printscreen in attachment.

![selection_002](https://cloud.githubusercontent.com/assets/5780150/24199695/e56eb184-0f0a-11e7-97ab-07bfaf35c9dc.png)

- Furthermore: at least the deduction amounts should be rounded at 0.05 cents [see the Swissdec Guidelines on page 20](https://www.swissdec.ch/fileadmin//user_upload/Software-Hersteller/RL4.0/RichtlinienLohndatenverarbeitung20130514_20130519_f.pdf)

open-net-sarl committed:
- Fixed "Payslip Computation Details" (Payslip line views) with wrong columns for "Rate" and "Amount"

open-net-sarl commented:
@didierdonze We used the Swissdec codes for the rules (**) but we dont want to follow completly all the Swissdec guidelines.

** because like that is easier to know which rule is, for exemple in french exists "AVS" but it doesnt in german, and we dont want to translate the codes.

open-net-sarl committed:
- Added new rule "Total Deduction"
- Added new rule category "Information" for the new rule
- Added traslations for the new rule and category
- Fixed UI rule not appearing in the payslip report

open-net-sarl committed:
- Fixed Rule 3030-VD not appearing in the payslip

open-net-sarl committed:
- Separated UI and UI SOL rules so on a migration nothing will crash
- Added new translations for UI SOL
- Created migration file to resolve the dependences and delete the old categories not used anymore

open-net-sarl committed:
- Resolved flake8 errors

leemannd commented on l10n_ch_hr_payroll/__manifest__.py:
```diff
@@ -40,16 +40,23 @@
         'account'
     ],
-    'version': '9.0.1.3.0',
+    'version': '10.0.1.0.0.0.3',
```
s/10.0.1.0.0.0.3/10.0.1.0.0/

leemannd commented on l10n_ch_hr_payroll/__init__.py:
```diff
@@ -1,5 +1,5 @@
-
-from . import models
+# -*- coding: utf-8 -*-
+# © 2012 David Coninckx (Opennet Sarl)
```
In the newest guidlines
s/©/Copyright/

open-net-sarl replied:
https://github.com/OCA/l10n-switzerland/blob/10.0/l10n_ch_base_bank/__manifest__.py

leemannd replied:
@open-net-sarl https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L2 & https://github.com/OCA/maintainer-tools/pull/197 
The one you are pointing to was left by mistake.

leemannd commented on l10n_ch_hr_payroll/__manifest__.py:
```diff
@@ -4,6 +4,7 @@
 #  Module: l10n_ch_hr_payroll
 #
 #  Created by sge@open-net.ch
+#  Updated by lfr@open-net.ch (2017)
```
Header for this file https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

leemannd commented on l10n_ch_hr_payroll/data/hr.salary.rule.category.xml:
```diff
@@ -4,10 +4,11 @@ File: hr.salary.rule.category.xml
 Module: l10n_ch_hr_payroll
 
 Created by sge@open-net.ch
+Updated by lfr@open-net.ch (2017)
```
Header change https://github.com/OCA/maintainer-tools/blob/master/template/module/views/model_name_view.xml#L2

leemannd commented on l10n_ch_hr_payroll/data/hr.salary.rule.xml:
```diff
@@ -4,616 +4,429 @@ File: hr.salary.rule.xml
 Module: l10n_ch_hr_payroll
 
 Created by sge@open-net.ch
+Updated by lfr@open-net.ch (2017)
```
https://github.com/OCA/maintainer-tools/blob/master/template/module/data/model_name_data.xml#L3

leemannd commented on l10n_ch_hr_payroll/data/hr.salary.rule.xml:
```diff
@@ -4,616 +4,429 @@ File: hr.salary.rule.xml
-    <data noupdate="1">
-        <record id="AM" model="hr.salary.rule">
+<odoo>
+    <data>
```
Why are you removing the 'noupdate' ?
If you ar e sure, you can remove also the '<data>'

leemannd commented on l10n_ch_hr_payroll/migrations/10.0.1.0.0.0.3/pre-update-rules.py:
```diff
@@ -0,0 +1,79 @@
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Short header

leemannd commented on l10n_ch_hr_payroll/models/__init__.py:
```diff
@@ -1,6 +1,12 @@
-from . import hr_employee
+# -*- coding: utf-8 -*-
+# © 2012 David Coninckx (Opennet Sarl)
+# © 2017 Leonardo Franja (Opennet Sarl)
```
s/©/Copyright/

leemannd commented on l10n_ch_hr_payroll/models/hr_contract.py:
```diff
@@ -1,16 +1,42 @@
-                              digits=dp.get_precision('Account'))
+# -*- coding: utf-8 -*-
+# © 2012 David Coninckx (Opennet Sarl)
+# © 2017 Leonardo Franja (Opennet Sarl)
```
s/©/Copyright/

leemannd commented on l10n_ch_hr_payroll/models/hr_employee.py:
```diff
@@ -1,15 +1,14 @@
-    children = fields.Integer('Number of Children at school')
-    children_student = fields.Integer('Number of Children student')
+# -*- coding: utf-8 -*-
+# © 2012 David Coninckx (Opennet Sarl)
```
s/©/Copyright/

leemannd commented on l10n_ch_hr_payroll/models/hr_employee.py:
```diff
@@ -1,15 +1,14 @@
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    # ---------- Fields management
```
? could be removed ?

leemannd commented on l10n_ch_hr_payroll/models/hr_payslip.py:
```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# © 2017 Leonardo Franja (Opennet Sarl)
```
s/©/Copyright/

leemannd commented on l10n_ch_hr_payroll/models/hr_payslip.py:
```diff
@@ -0,0 +1,70 @@
+# © 2017 Leonardo Franja (Opennet Sarl)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import openerp.addons.decimal_precision as dp
```
s/openerp/odoo/

leemannd commented on l10n_ch_hr_payroll/models/hr_employee.py:
```diff
@@ -1,15 +1,14 @@
+# © 2012 David Coninckx (Opennet Sarl)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import fields, models
```
s/openerp/odoo/

leemannd commented on l10n_ch_hr_payroll/models/hr_salary_rule.py:
```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# © 2017 Leonardo Franja (Opennet Sarl)
```
s/©/Copyright/

leemannd commented on l10n_ch_hr_payroll/models/hr_salary_rule_category.py:
```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+# © 2017 Leonardo Franja (Opennet Sarl)
```
s/©/Copyright/

leemannd commented on l10n_ch_hr_payroll/models/res_company.py:
```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+# © 2017 Leonardo Franja (Opennet Sarl)
```
s/©/Copyright/

leemannd commented on l10n_ch_hr_payroll/models/res_company.py:
```diff
@@ -0,0 +1,92 @@
+# © 2017 Leonardo Franja (Opennet Sarl)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import openerp.addons.decimal_precision as dp
```
s/openerp/odoo/

leemannd commented on l10n_ch_hr_payroll/tests/__init__.py:
```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
+# © 2017 Leonardo Franja (Open Net Sarl)
```
s/©/Copyright/

leemannd commented on l10n_ch_hr_payroll/tests/test_contract_lpp.py:
```diff
@@ -0,0 +1,261 @@
+# -*- coding: utf-8 -*-
+# © 2017 Leonardo Franja (Open Net Sarl)
```
s/©/Copyright/

leemannd commented on l10n_ch_hr_payroll/tests/test_contract_lpp.py:
```diff
@@ -0,0 +1,261 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+import openerp.tests.common as common
```
s/openerp/odoo/

leemannd commented on l10n_ch_hr_payroll/tests/test_max_lpp.py:
```diff
@@ -0,0 +1,226 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+import openerp.tests.common as common
```
s/openerp/odoo/

leemannd commented on l10n_ch_hr_payroll/tests/test_max_lpp.py:
```diff
@@ -0,0 +1,226 @@
+# -*- coding: utf-8 -*-
+# © 2017 Leonardo Franja (Open Net Sarl)
```
s/©/Copyright/

leemannd commented on l10n_ch_hr_payroll/tests/test_min_lpp.py:
```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+# © 2017 Leonardo Franja (Open Net Sarl)
```
s/©/Copyright/

leemannd commented on l10n_ch_hr_payroll/tests/test_min_lpp.py:
```diff
@@ -0,0 +1,225 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+import openerp.tests.common as common
```
odoo

leemannd commented on l10n_ch_hr_payroll/tests/test_no_lpp_rate.py:
```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+# © 2017 Leonardo Franja (Open Net Sarl)
```
s/©/Copyright/

leemannd commented on l10n_ch_hr_payroll/tests/test_wizard.py:
```diff
@@ -0,0 +1,215 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+import openerp.tests.common as common
```
odoo

leemannd commented on l10n_ch_hr_payroll/tests/test_wizard.py:
```diff
@@ -0,0 +1,215 @@
+# -*- coding: utf-8 -*-
+# © 2017 Leonardo Franja (Open Net Sarl)
```
s/©/Copyright/

leemannd commented on l10n_ch_hr_payroll/views/hr_contract_view.xml:
```diff
@@ -4,24 +4,33 @@
     Module: l10n_ch_hr_payroll
 
     Created by cyp@open-net.ch
+    Updated by lfr@open-net.ch (2017)
```
CHange the header please

leemannd commented on l10n_ch_hr_payroll/views/hr_contract_view.xml:
```diff
@@ -4,24 +4,33 @@
 -->
-<openerp>
+<odoo>
     <data>
```
If there is no 'noupdate' it can be removed

leemannd commented on l10n_ch_hr_payroll/views/hr_employee_view.xml:
```diff
@@ -7,7 +7,7 @@ Created by sge@open-net.ch
 -->
-<openerp>
+<odoo>
   <data>
```
to remove

leemannd commented on l10n_ch_hr_payroll/views/hr_payroll_config_view.xml:
```diff
@@ -0,0 +1,151 @@
+File: hr_payroll_config_view.xml
+Module: l10n_ch_hr_payroll
+
+Created by lfr@open-net.ch
```
Update

leemannd commented on l10n_ch_hr_payroll/views/hr_payroll_view.xml:
```diff
@@ -0,0 +1,41 @@
+    File: views/hr_payroll_view.xml
+    Module: l10n_ch_hr_payroll
+
+    Created by lfr@open-net.ch
```
Change

leemannd commented on l10n_ch_hr_payroll/views/hr_payroll_config_view.xml:
```diff
@@ -0,0 +1,151 @@
+Copyright (c) 2017 Open Net Sàrl
+-->
+<odoo>
+    <data>
```
to remove

leemannd commented on l10n_ch_hr_payroll/views/hr_payroll_view.xml:
```diff
@@ -0,0 +1,41 @@
+    Copyright (c) 2017 Open Net Sàrl
+-->
+<odoo>
+    <data>
```
To remove

leemannd commented on l10n_ch_hr_payroll/views/hr_payslip_line_view.xml:
```diff
@@ -0,0 +1,33 @@
+    File: views/hr_payslip_line_view.xml
+    Module: l10n_ch_hr_payroll
+
+    Created by lfr@open-net.ch
```
Update

leemannd commented on l10n_ch_hr_payroll/views/hr_payslip_line_view.xml:
```diff
@@ -0,0 +1,33 @@
+    Copyright (c) 2017 Open Net Sàrl
+-->
+<odoo>
+    <data>
```
Remove

leemannd commented on l10n_ch_hr_payroll/views/hr_payslip_view.xml:
```diff
@@ -0,0 +1,50 @@
+    File: views/hr_payslip_view.xml
+    Module: l10n_ch_hr_payroll
+
+    Created by lfr@open-net.ch
```
Update

leemannd commented on l10n_ch_hr_payroll/views/hr_payslip_view.xml:
```diff
@@ -0,0 +1,50 @@
+    Copyright (c) 2017 Open Net Sàrl
+-->
+<odoo>
+    <data>
```
remove

leemannd commented on l10n_ch_hr_payroll/views/hr_salary_rule_view.xml:
```diff
@@ -0,0 +1,30 @@
+File: hr_salary_rule_view.xml
+Module: l10n_ch_hr_payroll
+
+Created lfr@open-net.ch
```
Update

leemannd commented on l10n_ch_hr_payroll/views/lpp_contract_view.xml:
```diff
@@ -0,0 +1,25 @@
+    File: views/lpp_contract_view.xml
+    Module: l10n_ch_hr_payroll
+
+    Created by lfr@open-net.ch
```
Update

leemannd commented on l10n_ch_hr_payroll/views/lpp_contract_view.xml:
```diff
@@ -0,0 +1,25 @@
+    Copyright (c) 2017 Open Net Sàrl
+-->
+<odoo>
+    <data>
```
Remove

leemannd reviewed (changes requested):
Hello, thank you for the work.
No code review, no testing. Only nitpicking in order to fit OCA standards.

open-net-sarl committed:
- Changed code to fit OCA standards

leemannd commented on l10n_ch_hr_payroll/__init__.py:
```diff
@@ -1,5 +1,5 @@
-
-from . import models
+# -*- coding: utf-8 -*-
+# © 2017 Open Net Sàrl
```
still in all files © should be replaced.
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L2 & OCA/maintainer-tools#197

open-net-sarl replied:
Verified and fixed at the last commit.

leemannd commented on l10n_ch_hr_payroll/views/lpp_contract_view.xml:
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
```
If there is no noupdate, it should be removed. Can you check other files too?

open-net-sarl replied:
Verified and fixed at the last commit.

leemannd reviewed (changes requested):
Improved, but still a bit of change to do for fitting the guidelines.

open-net-sarl committed:
Changed menuitem (Payroll Configuration) to a better place:
- From : Payroll> Payroll Configuration
- To: Payroll>Configuration>Payroll Configuration

open-net-sarl committed:
Checked if salary rule exists in compute

open-net-sarl committed:
Added wage type to contract ( Hourly / Monthly )
- Added the 2 types of wage
- Corrected python computation for basic rule
- Added translations

open-net-sarl committed:
Resolved issues with flake8

open-net-sarl committed:
Fixed test issues

open-net-sarl committed:
Added more tests

open-net-sarl committed:
Corrected bad translation

open-net-sarl committed:
Added new rule "AS Families" (PC Familles):
- Added the rule
- Added translations
- Updated configuration wizard
- Updated views
- Updated tests

open-net-sarl commented:
```
ERROR: test_sct_ch_payment_type1 (odoo.addons.l10n_ch_pain_credit_transfer.tests.test_ch_sct.TestSCT_CH)

Traceback (most recent call last):

`   File "/home/travis/build/OCA/l10n-switzerland/l10n_ch_pain_credit_transfer/tests/test_ch_sct.py", line 141, in test_sct_ch_payment_type1

`     action = self.payment_order.open2generated()

`   File "/home/travis/dependencies/bank-payment/account_payment_order/models/account_payment_order.py", line 297, in open2generated

`     payment_file_str, filename = self.generate_payment_file()

`   File "/home/travis/dependencies/bank-payment/account_banking_sepa_credit_transfer/models/account_payment_order.py", line 112, in generate_payment_file

`     }, gen_args)

` TypeError: generate_start_payment_info_block() takes exactly 9 arguments (10 given)

ERROR: test_sct_ch_payment_type3 (odoo.addons.l10n_ch_pain_credit_transfer.tests.test_ch_sct.TestSCT_CH)

Traceback (most recent call last):

`   File "/home/travis/build/OCA/l10n-switzerland/l10n_ch_pain_credit_transfer/tests/test_ch_sct.py", line 224, in test_sct_ch_payment_type3

`     action = self.payment_order.open2generated()

`   File "/home/travis/dependencies/bank-payment/account_payment_order/models/account_payment_order.py", line 297, in open2generated

`     payment_file_str, filename = self.generate_payment_file()

`   File "/home/travis/dependencies/bank-payment/account_banking_sepa_credit_transfer/models/account_payment_order.py", line 112, in generate_payment_file

`     }, gen_args)

` TypeError: generate_start_payment_info_block() takes exactly 9 arguments (10 given)
```

As reported in the issue: https://github.com/OCA/l10n-switzerland/issues/335

open-net-sarl committed:
Fixed files header to match the OCA template
Removed unused "<data>" tags

open-net-sarl committed:
Fixed indentation in amount_python_compute at the salary rules

leemannd commented on l10n_ch_hr_payroll/data/hr.salary.rule.category.xml:
```diff
@@ -1,14 +1,5 @@
-Copyright (c) 2014 Open-Net Ltd.
--->
-<openerp>
-    <data noupdate="1">
```
Is there a reason to remove it?

open-net-sarl replied:
Everything must be updatable!

leemannd commented on l10n_ch_hr_payroll/data/hr.salary.rule.xml:
```diff
@@ -1,619 +1,471 @@
-Copyright (c) 2015 Open Net Sàrl
--->
-<openerp>
-    <data noupdate="1">
```
Is there a reason to remove it?

open-net-sarl replied:
Everything must be updatable!

leemannd reviewed (changes requested):
On the formal side, LGTM. Not tested.
Only two questions pending.
Thank you for the work.

open-net-sarl committed:
Added "100"% as default at the working rate of the payslip

open-net-sarl committed:
Renamed rule "Family Allowances (VD)" to "Family Allowances":
- Added 4 new fields to help computing the rule
- Added translations
- Corrected the test files

Organized the configuration wizard:
- Created new page to hold the values alied to cantonal rules

open-net-sarl committed:
Fixed flake8 errors and fixed module tests

open-net-sarl committed:
Corrected variable name

open-net-sarl committed:
Fixed configuration view style
Fixed french translation

open-net-sarl committed:
Corrected family allowance fields names and translations

didierdonze reviewed (approved):
Ok, from a business perspective. Can it be merged?

open-net-sarl commented:
@didierdonze 
For the author, its ok to merge this.
